doc/source/guide/citing.rst

```diff
@@ -0,0 +1,140 @@
+of that commit.
+
+Automatic Citation Collection
+-----------------------------
+
+QInfer also supports the use of the `duecredit`_ project to automatically
+collect citations for a bibliography. This support is still experimental,
+and may not yet generate complete bibliographies, so please manually check
+the resulting bibliography. In any case, to get started, install `duecredit`_::
+
+    $ pip install duecredit
+
+If your project then uses a script to generate and/or analyze data, then you
+can use `duecredit`_ to collect citations for Python modules and functions
+called by your script. For example, if your project can be run as
+`python script.py`, use the following command to collect a bibliography::
+
+    $ python -m duecredit script.py
+
+Alternatively, it may be more reliable to use environment variables to turn
+on `duecredit`_ collection, since this approach also works with Jupyter Notebook::
+
+    $ export DUECREDIT_ENABLE=yes # Bash
+    PS> $Env:DUECREDIT_ENABLE = "yes" # PowerShell
+
+In either case, this will create a file called ``.duecredit.p`` containing a representation
+of your bibliography. To print it out in BibTeX form, use the summary functionality
+of `duecredit`_::
+
+    $ duecredit summary --format=bibtex
+
+Note that this summary will also include projects such as NumPy and SciKit-Learn
+that are supported by `duecredit`_, as well as any other projects which natively
+host citation metadata through `duecredit`_.  For example::
+
+    $ export DUECREDIT_ENABLE=yes
+    $ ipython
+    In [1]: import qinfer as qi
+    In [2]: exit
+
+    DueCredit Report:
+    - Scientific tools library / numpy (v 1.11.1) [1]
+    - Bayesian inference for quantum information / qinfer (v 1.0) [2]
+    - Machine Learning library / sklearn (v 0.17.1) [3]
+        - Affinity propagation clustering algorithm / sklearn.cluster.affinity_propagation_ (v 0.17.1) [4]
+
+    3 packages cited
+    1 module cited
+    0 functions cited
+
+    References
+    ----------
+
+    [1] Van Der Walt, S., Colbert, S.C. & Varoquaux, G., 2011. The NumPy array: a structure for efficient numerical computation. Computing in Science & Engineering, 13(2), pp.22–30.
+    [2] Granade, C. et al., 2016. QInfer: Statistical Inference Software for Quantum Applications. arXiv:1610.00336 [physics, physics:quant-ph, stat].
+    [3] Pedregosa, F. et al., 2011. Scikit-learn: Machine learning in Python. The Journal of Machine Learning Research, 12, pp.2825–2830.
+    [4] Frey, B.J. & Dueck, D., 2007. Clustering by Passing Messages Between Data Points. Science, 315(5814), pp.972–976.
+
+The bibliography entries defined by QInfer are organized according to different
+*tags*, making it easier to filter through the results of `duecredit`_. In particular,
+QInfer defines the following citation tags:
+
+- ``implementation``: Concerns a software implementation of an algorithm or computation.
+- ``theory``: Concerns the initial proposal of an algorithm and other theory results.
```

Comment:
interesting... so how this is different from implementation really?  is it when you are to provide a "reference-implementation" of a new method (may be not yet published, thus may be no reference even yet) which rests on this "theory" publication, so not implementing but rather extending etc upon it?

meanwhile I've decided to accumulate those tags within our README.md: https://github.com/duecredit/duecredit/pull/109 . Feel welcome to chime in ;)